### PR TITLE
set `DepositUpdate` sweep tx fee in `handle_completed_deposit` 

### PR DESCRIPTION
**Pull request by Jiloc**
## Description

Closes: #722

## Changes
- Introduced a new `get_deposit_request` query.
- Updated `handle_completed_deposit` to fetch the deposti request connected to the event and compute the fee as the difference between the deposited amount and the amount of sBTC minted.
 
## Testing Information
- adapted existing unit and integration tests to work with the new change
- added new unit test and integration tests

### CODE REVIEW

**Review by djordon — COMMENTED** [review]
Nice, I think we can simplify this a bit. If we subtract the amount in the `CompletedDepositEvent` from the deposit request amount then that will give us the fee.

**Comment by cylewitruk** [comment]
I've kinda been thinking in the lines of removing the sweep tables since we have the information elsewhere (i.e. in the bitcoin_* tables). This creates a new dependency on them -- can we acoid that?

**Comment by djordon** [comment]
> the sweep tables since we have the information elsewhere (i.e. in the bitcoin_* tables). This creates a new dependency on them -- can we avoid that?

Yeah, we should only need the `deposit_requests` table here.

File: signer/src/error.rs [hunk 1]
```diff
@@ -26,6 +26,10 @@ pub enum Error {
     #[error("sweep transaction not found: {0}")]
     MissingSweepTransaction(bitcoin::Txid),
 
+    /// Indicates that a deposit request with the specified txid and vout could not be found.
+    #[error("deposit request not found: {0}, {1}")]
+    MissingDepositRequest(bitcoin::Txid, u32),
```
Comment (djordon) [inline signer/src/error.rs:31]: Use the `OutPoint` type here, it's simpler.

File: signer/tests/integration/stacks_events_observer.rs [hunk 2]
```diff
@@ -415,6 +421,20 @@ async fn test_new_blocks_sends_update_deposits_to_emily() {
...
+    sweep.txid = deposit_completed_event.sweep_txid.into();
+    sweep.created_at_block_hash = deposit_completed_event.sweep_block_hash.into();
+    // A real sweep transaction would have the relevant deposits. For this test, we don't need them.
+    sweep.swept_deposits = vec![];
+    sweep.swept_withdrawals = vec![];
+    context
+        .get_storage_mut()
+        .write_sweep_transaction(&sweep)
+        .await
+        .expect("failed to insert dummy sweep transaction");
```
Comment (djordon) [inline signer/tests/integration/stacks_events_observer.rs:436]: Can't this all be removed?

**Review by djordon — APPROVED** [review]
Looks good, had a few nits.

File: signer/src/error.rs (hunk 1 above)
Comment (Jiloc) [reply signer/src/error.rs:31]: done in https://github.com/stacks-network/sbtc/pull/956/commits/c803d9d18bc48fc3aae449ca3c59233a26f44228

File: signer/tests/integration/stacks_events_observer.rs (hunk 2 above)
Comment (Jiloc) [reply signer/tests/integration/stacks_events_observer.rs]: yeah, I had to swap it with a deposit request 🤦 . Done in https://github.com/stacks-network/sbtc/pull/1005/commits/2b6e492f8f4a0868f541cf59b8f33bec06c25e67

File: signer/src/error.rs (hunk 1 above)
Comment (Jiloc) [reply signer/src/error.rs]: changed in https://github.com/stacks-network/sbtc/pull/1005/commits/23b091cb0204cde75f0f5f0d96662a983b6febe0

**Review by djordon — APPROVED** [review]
Looks good! 🚢

**Pull request merged**